### PR TITLE
feat: uv support, py 3.13

### PR DESCRIPTION
**Pull request by iloveitaly**
Curious what you think of this approach to supporting uv. Happy to add some tests for this as well, wanted to check on the approach before I built out the tests.

Fixes: https://github.com/railwayapp/nixpacks/issues/1128

### CODE REVIEW

**Review by JakeCooper — COMMENTED** [review]
I mean this looks pretty good to me! Wanna add some tests and a release label?

File: src/providers/python.rs [hunk 1]
```diff
@@ -92,13 +96,40 @@ impl Provider for PythonProvider {
...
+            plan.add_variables(EnvironmentVariables::from([(
+                "NIXPACKS_UV_VERSION".to_string(),
+                version,
+            )]));
+
+            plan.add_variables(EnvironmentVariables::from([(
+                "UV_PROJECT_ENVIRONMENT".to_string(),
+                VENV_LOCATION.to_string(),
+            )]));
+        }
```
Comment (coffee-cup) [inline src/providers/python.rs:131]: suggestion: you can add these in a single command

**Comment by iloveitaly** [comment]
Alright, build is finally passing here. I needed to make some additional updates to properly support the latest nixpkgs version (which included py 3.13).

File: src/providers/python.rs [hunk 2]
```diff
@@ -247,6 +297,12 @@ impl PythonProvider {
             ))));
         }
 
+        // the python package is extracted from pyproject.toml, but this can often not be the desired entrypoint
+        // for this reason we prefer main.py to the module heuristic used in the pyproject.toml logic
```
Comment (coffee-cup) [inline src/providers/python.rs:301]: This is nice 👍

**Review by coffee-cup — APPROVED** [review]
Thanks for this feature and uv support!

**Pull request merged**